pkg/plugin/client/client.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package client
+
+import (
+	"fmt"
+	"os/exec"
+
+	hclog "github.com/hashicorp/go-hclog"
+	goplugin "github.com/hashicorp/go-plugin"
+
+	"aspect.build/cli/pkg/ioutils"
+	"aspect.build/cli/pkg/plugin/loader"
+	"aspect.build/cli/pkg/plugin/sdk/v1alpha2/config"
+	"aspect.build/cli/pkg/plugin/sdk/v1alpha2/plugin"
+)
+
+// Factory hides the call to goplugin.NewClient.
+type Factory interface {
+	New(config loader.AspectPlugin, streams ioutils.Streams) (PluginInstance, error)
+}
+
+func NewFactory() Factory {
+	return &clientFactory{}
+}
+
+type clientFactory struct{}
+
+// New calls the goplugin.NewClient with the given config.
+func (*clientFactory) New(aspectplugin loader.AspectPlugin, streams ioutils.Streams) (PluginInstance, error) {
+	var res PluginInstance
```

Comment:
This is created just about only so it can be used as a "type nil" value (is that the correct terminology?) when returning an error.